libs/cohere/langchain_cohere/chat_models.py

```diff
@@ -52,6 +53,40 @@
 from langchain_cohere.llms import BaseCohere
 
 
+def _get_tool_results(messages: List[BaseMessage]) -> List[Dict[str, Any]]:
+    """Get tool_results from messages."""
+    tool_results = []
+    for message in messages:
+        if isinstance(message, ToolMessage):
+            tool_message = message
+            previous_ai_msgs = [
+                message
+                for message in messages[:-1]
+                if isinstance(message, AIMessage) and message.tool_calls
+            ]
+            if previous_ai_msgs:
+                previous_ai_msg = previous_ai_msgs[-1]
+                tool_message_name: Optional[str] = None
+                if tool_message.name:
+                    tool_message_name = tool_message.name
+                else:
+                    tool_message_name = tool_message.additional_kwargs.get("name")
+                tool_results.extend(
+                    [
+                        {
+                            "call": ToolCall(
+                                name=lc_tool_call["name"],
+                                parameters=lc_tool_call["args"],
```

Comment:
The parameters should match those that are sent with `bind_tools` - the logic is in `_convert_to_cohere_tool` but I'd be more than happy to extract it!

Reply:
I may be misunderstanding here. These are calls to tools (i.e., specifying parameter values), instead of definitions of tool schemas. How should we use `_convert_to_cohere_tool` here?

Reply:
I was mistaken, apologies! I thought that we converted the arguments (and that some of the logic in _convert_to_cohere_tool converted back), but I was mistaken